File: _정답코드/4주차/05_쿼드트리_이미지_복원/Solution.java
public class Solution {
    public static int[][] answer = new int[8][8];
    public static String comp;
    public static int idx = 0;

	public int[][] solution(String compressed) {
        comp = compressed;
        decode(0, 0, 8);
		return answer;
	}

    private void decode(int x, int y, int n){
        if( comp.charAt(idx) != '(' ){
            int value = Integer.parseInt(comp.charAt(idx)+"");
            
            for(int i=x; i<x+n; i++){
                for(int j=y; j<y+n; j++){
                    answer[i][j] = value;
                }
            }
            
            idx ++;
        } else {
            idx ++; // 여는 괄호 생략
            
            decode(x, y, n/2);
            decode(x, y+n/2, n/2);
            decode(x+n/2, y, n/2);
            decode(x+n/2, y+n/2, n/2);
            
            idx ++; // 닫는 괄호 생략
        }
    }
}

File: _정답코드/4주차/05_쿼드트리_이미지_복원/solution.cpp
#include <vector>
#include <string>
using namespace std;

vector<vector<int>> answer(8, vector<int>(8));
string comp;
int idx = 0;

void decode(int x, int y, int n) {
    if ( comp[idx] != '(' ) {
        int value = comp[idx] - '0';
        for (int i = x; i < x + n; i++) {
            for (int j = y; j < y + n; j++) {
                answer[i][j] = value;
            }
        }
        idx += 1;
    } else {
        idx += 1;
        decode(x, y, n / 2);
        decode(x, y + n / 2, n / 2);
        decode(x + n / 2, y, n / 2);
        decode(x + n / 2, y + n / 2, n / 2);
        idx += 1;
    }
}

vector<vector<int>> solution(string compressed) {
    comp = compressed;
    idx = 0;
    decode(0, 0, 8);
    return answer;
}

